core/src/main/java/hudson/cli/DeleteNodeCommand.java

```diff
@@ -78,7 +78,7 @@ protected int run() throws Exception {
                 }
 
                 node.toComputer().doDoDelete();
-            } catch (AccessDeniedException e) {
+            } catch (AccessDeniedException2 e) {
```

Comment:
@olivergondza 
e.toString() says:
hudson.security.AccessDeniedException2: user is missing the Slave/Delete permission

Note: I caught that during the work on connect-node (PR#1923).